### PR TITLE
[WIP] Add GPU plugin in the WebUI

### PR DESCRIPTION
**Pull request by notFloran**
#### Description

Integrate the GPU plugin in the WebUI.

#### Resume

* Bug fix: no
* New feature: yes
* Fixed tickets: #170

### CODE REVIEW

**Comment by notFloran** [comment]
Don't merge the MR !

There is still a display issue with the the top menu, I'll see with @nclsHart to fix it.

@nicolargo Can you test this first version ?

**Comment by nicolargo** [comment]
@notFloran Here are my tests results:

[{"key": "gpu_id", "mem": None, "proc": 60, "gpu_id": 0, "name": "GeForce GTX 560 Ti"}]

selection_283

=> Ok

[{"key": "gpu_id", "mem": 10, "proc": 60, "gpu_id": 0, "name": "GeForce GTX 560 Ti"}]

selection_284

=> Ok

[{"key": "gpu_id", "mem": 48.64645, "proc": 60.73, "gpu_id": 0, "name": "GeForce GTX 560 Ti"},
 {"key": "gpu_id", "mem": 70.743, "proc": 80.28, "gpu_id": 1, "name": "GeForce GTX 560 Ti"},
 {"key": "gpu_id", "mem": 0, "proc": 0, "gpu_id": 2, "name": "GeForce GTX 560 Ti"}]

selection_285

=> Not OK (only one line should be display for the header)

and in meangpu mode:

selection_286

=> Not OK (only one line should be display for the header). The % should be on the same line.

[{"key": "gpu_id", "mem": 48.64645, "proc": 60.73, "gpu_id": 0, "name": "GeForce GTX 560 Ti"},
 {"key": "gpu_id", "mem": None, "proc": 80.28, "gpu_id": 1, "name": "GeForce GTX 560 Ti"},
 {"key": "gpu_id", "mem": 0, "proc": 0, "gpu_id": 2, "name": "ANOTHER GPU"}]

selection_287

=> Not OK (only one line should be display for the header)

and in meangpu mode:

selection_288

=> Not OK (only one line should be display for the header). The % should be on the same line.

You can test it on your side by uncomment the self.stats var in the glances/plugins/glances_gpu.py file:

```
        # !!! JUST FOR TEST
        # self.stats = [{"key": "gpu_id", "mem": None, "proc": 60, "gpu_id": 0, "name": "GeForce GTX 560 Ti"}]
        # self.stats = [{"key": "gpu_id", "mem": 10, "proc": 60, "gpu_id": 0, "name": "GeForce GTX 560 Ti"}]
        # self.stats = [{"key": "gpu_id", "mem": 48.64645, "proc": 60.73, "gpu_id": 0, "name": "GeForce GTX 560 Ti"},
        #               {"key": "gpu_id", "mem": 70.743, "proc": 80.28, "gpu_id": 1, "name": "GeForce GTX 560 Ti"},
        #               {"key": "gpu_id", "mem": 0, "proc": 0, "gpu_id": 2, "name": "GeForce GTX 560 Ti"}]
        # self.stats = [{"key": "gpu_id", "mem": 48.64645, "proc": 60.73, "gpu_id": 0, "name": "GeForce GTX 560 Ti"},
        #               {"key": "gpu_id", "mem": None, "proc": 80.28, "gpu_id": 1, "name": "GeForce GTX 560 Ti"},
        #               {"key": "gpu_id", "mem": 0, "proc": 0, "gpu_id": 2, "name": "ANOTHER GPU"}]
        # !!! TO BE COMMENTED
```

Do not forget to comment all the line before the pull request ;)

**Comment by nicolargo** [comment]
@notFloran Can i merge this PR ?

File: glances/outputs/static/html/plugins/gpu.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        <div class="table-cell"></div>
+    </div>
+    <div class="table-row" ng-if="arguments.meangpu || statsGpu.gpus.length === 1">
+        <div class="table-cell text-left">proc:</div>
+        <div class="table-cell" ng-class="statsGpu.getMeanDecoration('proc')" ng-if="statsGpu.mean.proc">{{ statsGpu.mean.proc | number : 0 }} %</div>
+        <div class="table-cell" ng-if="!statsGpu.mean.proc">N/A</div>
+    </div>
+    <div class="table-row" ng-if="arguments.meangpu || statsGpu.gpus.length === 1">
+        <div class="table-cell text-left">mem:</div>
+        <div class="table-cell" ng-class="statsGpu.getMeanDecoration('mem')" ng-if="statsGpu.mean.mem">{{ statsGpu.mean.mem | number : 0 }} %</div>
```
Comment (nclsHart) [inline glances/outputs/static/html/plugins/gpu.html]: remove space before `%`

File: glances/outputs/static/html/plugins/gpu.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<div class="table">
+    <div class="table-row">
+        <div class="table-cell text-left title">{{ statsGpu.name }}</div>
+        <div class="table-cell"></div>
+    </div>
+    <div class="table-row" ng-if="arguments.meangpu || statsGpu.gpus.length === 1">
+        <div class="table-cell text-left">proc:</div>
+        <div class="table-cell" ng-class="statsGpu.getMeanDecoration('proc')" ng-if="statsGpu.mean.proc">{{ statsGpu.mean.proc | number : 0 }} %</div>
```
Comment (nclsHart) [inline glances/outputs/static/html/plugins/gpu.html]: remove space before `%`

File: glances/outputs/static/html/stats.html [hunk 3]
```diff
@@ -29,6 +29,9 @@
         <div class="col-sm-12 col-md-8 col-lg-6" ng-if="!arguments.disable_cpu && arguments.percpu">
             <section id="per_cpu" class="plugin" ng-include src="'plugins/per_cpu.html'"></section>
         </div>
+        <div class="hidden-xs hidden-sm hidden-md col-lg-3" ng-if="!arguments.disable_gpu && statsGpu.gpus.length > 0">
```
Comment (nclsHart) [inline glances/outputs/static/html/stats.html]: The easier way that I see to have enough space for this plugin is to hide the mem_more part when this plugin is displayed. You can do this by using the following code for the ng-if of mem_more : 

`!arguments.disable_mem && statsGpu.gpus.length = 0`

If you are ok with solution, you need to display the gpu plugin in medium(md) size too:

`hidden-xs hidden-sm col-md-4 col-lg-3`

**Comment by nclsHart** [comment]
You also need to edit `help.html` to add the gpu shorcut.

**Comment by notFloran** [comment]
@nicolargo For the top menu issue I implemented the solution proposed by @nclsHart, are you ok with this ?

> The easier way that I see to have enough space for this plugin is to hide the mem_more part when this plugin is displayed. 

If you are ok with this solution, I think we can merge.

**Comment by nicolargo** [comment]
@notFloran : Agree, it is exactly what is it done in the Curses interface.

**Comment by nclsHart** [comment]
@nicolargo be careful, the shortcut is missing from the help page in both interfaces.

**Comment by nicolargo** [comment]
@nclsHart Ok, waiting for another PR with the shortcut. I merge others commits in order to start the validation on my side.

**Pull request merged**